src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -59,7 +59,7 @@ def list_role_definitions(cmd, name=None, resource_group_name=None, scope=None,
                           custom_role_only=False):
     definitions_client = _auth_client_factory(cmd.cli_ctx, scope).role_definitions
     scope = _build_role_scope(resource_group_name, scope,
-                              definitions_client.config.subscription_id)
+                              definitions_client._config.subscription_id)
```

Comment:
Track 2 now makes `config` a protected attribute as `_config`. I can't find a better way to **extract subscription ID** from the SDK client.

Another option is to let `get_mgmt_service_client` return the subscription ID during client creation:

https://github.com/Azure/azure-cli/blob/21266e02457263420b62fba1c529f69e28852414/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L50-L51

but this is such a big breaking change that it will affect almost all command modules and extension.

Reply:
This pattern has already been used by `resource` command module:

https://github.com/Azure/azure-cli/blob/1b62d205ad1ce8aa56075fbafb9a409eeb1c1d04/src/azure-cli/azure/cli/command_modules/resource/custom.py#L3551-L3552

Reply:
This change was mentioned in doc: https://github.com/Azure/azure-cli/blob/dev/doc/track_2_migration_guidance.md#obtaining-subscription

However, the doc is not accurate as the subscription used to create the client may not be the same as the current subscription. Assuming the subscription used to create the client being the same as the current subscription can introduce bugs very difficult to track down.